dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/SingleDataRequestManager.kt

```diff
@@ -38,20 +36,34 @@ class SingleDataRequestManager(
     @Transactional
     fun processSingleDataRequest(singleDataRequest: SingleDataRequest): List<StoredDataRequest> {
         utils.throwExceptionIfNotJwtAuth()
-        dataRequestLogger.logMessageForSingleDataRequestReceived()
+        dataRequestLogger.logMessageForReceivingSingleDataRequest(singleDataRequest.companyIdentifier)
         validateContactsAndMessage(singleDataRequest.contacts, singleDataRequest.message)
-        val (identifierTypeToStore, identifierValueToStore) = identifyIdentifierTypeAndTryGetDatalandCompanyId(
-            singleDataRequest.companyIdentifier,
+        var datalandCompanyId: String?
+
+        dataRequestLogger.logMessageForReceivingSingleDataRequest(singleDataRequest.companyIdentifier)
+        if (companyIdRegex.matches(singleDataRequest.companyIdentifier)) {
+            checkIfCompanyIsValid(singleDataRequest.companyIdentifier)
+            datalandCompanyId = singleDataRequest.companyIdentifier
+        } else {
```

Comment:
I can follow the logic here. the else case is for when the requested id does not match the companyIdRegex, which means it naturally can't ever be matched with an existing companyId.

Reply:
In that case it is no dataland company id but maybe still and isin or lei. In that case we want to search the dataland company id for the company with that lei/isin. utils.getDatalandCompanyIdForIdentifierValue can not be called for dataland companyIds I think.